### ISSUE TITLE
Структуры данных

### ISSUE THREAD

**Issue by MinkoJ2** [MinkoJ2/Related-To-Testing-Article#2]
Вопрос по задачам:

https://learn.javascript.ru/object-for-in#tasks - задачи 3/4

3. Готовый код - не понятно как выводится имя

"use strict";

var salaries = {
  "Вася": 100,
  "Петя": 300,
  "Даша": 250
};

var max = 0;
var maxName = "";
for (var name in salaries) {
  if (max < salaries[name]) {
    max = salaries[name];
    maxName = name;
  }
}

alert( maxName || "нет сотрудников" );

4. Готовый код - не понятно как пропустить умножение если там строка
var menu = {
  width: 200,
  height: 300,
  title: "My menu"
};

function isNumeric(n) {
  return !isNaN(parseFloat(n)) && isFinite(n);
}

function multiplyNumeric(obj) {
  for (var key in obj) {
    if (isNumeric(obj[key])) {
      obj[key] *= 2;
    }
  }
}

multiplyNumeric(menu);

alert( "menu width=" + menu.width + " height=" + menu.height + " title=" + menu.title );

@atherdon

**Comment by atherdon** [comment]
3. ну здесь name это название поля. но, т.к. типы данных в JS не строгие - то получается что объект можно обходить также как и массив. тоесть salaries[0] вернут зп первого сотрудника. а вот если ты пройдешься циклом - тогда, index == 'Vasya', value == 100

4. ну тут ты чето совсем сильно намудрил в проверке isNumeric. код сложно читается и я сразу нифига не понял что он делает. лучше сделай switch или несколько if - это будет проще понимать и может ты тогда не запутаешься. конечно хорошо что ты попытался все предусмотреть, но лучше делай более пошагово. вот как бы делал я

--закоментировал бы твой ад в этой функции.
поставил бы одну проверку, например `isFinite` и для каждого значения в цикле я бы еще и выводил данные алертом. ну чтобы было понятно для какого именно элемента срабатывает.

потом бы добавил еще еще проверку и опять все прогнал. будь больше параноиком. пытайся понимать что код сразу написанный работать не будет. и лучше пошагово идти а не сразу в лоб.

если чето будет непонятно - сразу маякуй

**Comment by MinkoJ2** [comment]
Код писал не я, это был код уже решенных задач 
@atherdon

**Comment by atherdon** [comment]
а есть задачи чтобы ты сам ручками пописал? я могу тебе выдать парочку -
мне не сложно. могу и абстракнтую выдать и связанную с проектом, заодно
потыкаешь как я все писал


On 07-Jul-2018 at 03:03 PM, MinkoJ2 wrote:


Код писал не я, это был код уже решенных задач
@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403221825>,
or mute the thread

**Comment by MinkoJ2** [comment]
Выдай парочку, потренируюсь 
@atherdon

**Comment by atherdon** [comment]
ну первая задача логичная - сделать 3хмерную вложеность объектов. и
погонять по их полям.
если изучил методы внутри объектов - можно и их поиспользовать.

создать объект College -> Groups -> People
у каждого объекта получается еще и свои внутренние поля должны быть. причем
получается что в колледже у тебя массив групп(объектов), а в каждой группе
у тебя массив учеников

пингани когда второе задание появится и заодно в комменте скинешь мне
ссылку. Важно создавать это все в новом файле, чтобы ты старое не затирал


On 07-Jul-2018 at 03:37 PM, MinkoJ2 wrote:


Выдай парочку, потренируюсь
@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403223818>,
or mute the thread

**Comment by MinkoJ2** [comment]
Как реализовать объект в объекте???
@atherdon

**Comment by atherdon** [comment]
`
const car = {
engine: {}
}; `
но это немного не то что в задаче - я просто на твой вопрос ответил


On 07-Jul-2018 at 04:16 PM, MinkoJ2 wrote:


Как реализовать объект в объекте???
@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403226261>,
or mute the thread

**Comment by MinkoJ2** [comment]
```
var collage = {
	name: 'ПТНГУ',
	groups: {
		group: 'КС-1-17',
	}
};

alert(groups.group); // не реагирует
```

**Comment by atherdon** [comment]
пиши данные внутри только на англ, а то мне глаз режет 6)
ну у тебя главная переменная не так называется. поэтому и не реагирует.
опять решил сразу вывести. иди мелкими шагами


On 07-Jul-2018 at 05:34 PM, MinkoJ2 wrote:


var collage = {
	name: 'ПТНГУ',
	groups: {
		group: 'КС-1-17',
	}
};

alert(groups.group); // не реагирует

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403231523>,
or mute the thread

**Comment by MinkoJ2** [comment]
Сделал (смот. folder_2)
@atherdon

**Comment by atherdon** [comment]
@MinkoJ2 ни ненада :)

перечитай еще раз задание, и напиши мне - где именно и что ты сделал неправильно. и что по твоему мнению нужно изменить/исправить/удалить/улучшить...

**Comment by atherdon** [comment]
задание это не простое. подумай хорошо. а то усложню

**Comment by MinkoJ2** [comment]
> @MinkoJ2 ни ненада :)
перечитай еще раз задание, и напиши мне - где именно и что ты сделал неправильно. и что по твоему мнению нужно изменить/исправить/удалить/улучшить...

Глянуть что не так в моём коде и что по моему мнению можно там изменить?
@atherdon

**Comment by MinkoJ2** [comment]
[Кликни, и я пошёл спать](https://github.com/MinkoJ2/Related-To-Testing-Article/tree/master/Objects)
@atherdon

**Comment by atherdon** [comment]
да, уже лучше выглядит. но это не финальный результат. ты все еще теряешь важный момент.
нехочу тебя мучать, но тебе надо добавить в объект несколько групп со всеми нужными данными.
сейчас я вижу что ты сделал везде тройную вложенность, но при этом у тебя всегда только один класс/группа представлен.

1) добавь в поле `class`, объекта `school` еще парочку классов
2) тут вроде ты нащупал что я хочу. но при этом, ты разделил данные на несколько частей. собери все вместе. 
- добавь все в одни объект.
- сделай так чтобы несколько групп было внутри колледжа. и внеси данные, которые относятся к группе. представь себе что у тебя есть например 3 группы. и в каждой группе, каждому студенту нужно выдать задание с уникальным номером. как ты сделаешь это с текущей структурой - никак

**Comment by MinkoJ2** [comment]
> сделай так чтобы несколько групп было внутри колледжа. и внеси данные, которые относятся к группе. представь себе что у тебя есть например 3 группы. и в каждой группе, каждому студенту нужно выдать задание с уникальным номером. как ты сделаешь это с текущей структурой - никак

Ты имеешь ввиду каждому ученику дать свой номер?
@atherdon

**Comment by atherdon** [comment]
дай каждому ученику отдельный номер
дай каждой группе отдельный номер


On 09-Jul-2018 at 05:56 AM, MinkoJ2 wrote:


сделай так чтобы несколько групп было внутри колледжа. и внеси данные,
которые относятся к группе. представь себе что у тебя есть например 3
группы. и в каждой группе, каждому студенту нужно выдать задание с
уникальным номером. как ты сделаешь это с текущей структурой - никак

Ты имеешь ввиду каждому ученику дать свой номер?
@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403367961>,
or mute the thread

**Comment by atherdon** [comment]
```
const object = {
   field: [1, 2, 3, 4, 5]
} 
```
и если мы говорим о присвоении номера - то он должен быть ключем в массиве, а не частью строки.

**Comment by MinkoJ2** [comment]
Если я сделаю так код, он не будет работать
```
alert(groupList);
var x = prompt('Enter number group', 0);

var college = {
	group : ['KS-1-17', 'DV-1-17', 'BO-1-17', 'OP-1-17'],
  	groupList : ['№ | Group\n' + '0. KS-1-17\n' + '1. DV-1-17\n' + '2. BO-1-17\n' + '3. OP-1-17'], // Присвоил каждой группе номер
	members : [
		{name: '0. Bogach Eduard\n' + '1. Egor Badanin\n' + '2. Igor Bantyk'}, // Присвоил каждому студенту номер
		{name: '0. Anny Chabanec\n' + '1. Karina Klimova\n' + '2. Kristina Shevchenko'}, // Присвоил каждому студенту номер
		{name: '0. Lesya Ivanova\n' + '1. Viktoria Lykyanenko\n' + '2. Kirill Ivanov'}, // Присвоил каждому студенту номер
		{name: '0. Emil Petrov\n' + '1. Egor Ivanov\n' + '2. Sergey Durnin'}] // Присвоил каждому студенту номер
	groups : {
		name : 'Group name: \n' + group[x],
		people : {
			name : members[x].name
		}
	}
};

alert(college.groups.name)
alert(college.groups.people.name)
```
@atherdon

**Comment by atherdon** [comment]
сейчас группы и студенты этих групп неособо соединены. но уже близко.
почему бы для групп не сделать такую сложенность
`
[Группы][0][имя *группы][студенты][0][имя*студента]
`


On 09-Jul-2018 at 06:53 AM, MinkoJ2 wrote:


Если я сделаю так код, он не будет работать

alert(groupList);
var x = prompt('Enter number group', 0);

var college = {
	group : ['KS-1-17', 'DV-1-17', 'BO-1-17', 'OP-1-17'],
  	groupList : ['№ | Group\n' + '0. KS-1-17\n' + '1. DV-1-17\n' + '2.
BO-1-17\n' + '3. OP-1-17'], // Присвоил каждой группе номер
	members : [
		{name: '0. Bogach Eduard\n' + '1. Egor Badanin\n' + '2. Igor
Bantyk'}, // Присвоил каждому студенту номер
		{name: '0. Anny Chabanec\n' + '1. Karina Klimova\n' + '2. Kristina
Shevchenko'}, // Присвоил каждому студенту номер
		{name: '0. Lesya Ivanova\n' + '1. Viktoria Lykyanenko\n' + '2.
Kirill Ivanov'}, // Присвоил каждому студенту номер
		{name: '0. Emil Petrov\n' + '1. Egor Ivanov\n' + '2. Sergey
Durnin'}] // Присвоил каждому студенту номер
	groups : {
		name : 'Group name: \n' + group[x],
		people : {
			name : members[x].name
		}
	}
};

alert(college.groups.name)
alert(college.groups.people.name)

@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403377920>,
or mute the thread

**Comment by MinkoJ2** [comment]
Ты предлагаешь сделать массив в массиве? @atherdon

**Comment by MinkoJ2** [comment]
Сделал пока что только такой вариант, как ты хочет 
```
var group = [
{gr : 'KS-1-17', mem : 'Bogach Eduard'}, 
{gr : 'DV-1-17', mem : '0. Anny Chabanec\n' + '1. Karina Klimova\n' + '2. Kristina Shevchenko'}, 
{gr : 'BO-1-17', mem : '0. Lesya Ivanova\n' + '1. Viktoria Lykyanenko\n' + '2. Kirill Ivanov'}, 
{gr : 'OP-1-17', mem : '0. Emil Petrov\n' + '1. Egor Ivanov\n' + '2. Sergey Durnin'}]; 
var groupList = ['№ | Group\n' + '0. KS-1-17\n' + '1. DV-1-17\n' + '2. BO-1-17\n' + '3. OP-1-17']; // Присвоил каждой группе номер

alert(groupList);
var x = prompt('Enter number group', 0);

 var college = {
	groups : {
		name : 'Group name: \n' + group[x].gr,
		people : {
			name : 'Members: \n' + group[x].mem
		}
	}
}; 

alert(college.groups.name)
alert(college.groups.people.name)
```
@atherdon

**Comment by atherdon** [comment]
т.к. я тебе помогу - прийдется еще сильнее усложнить эту задачу

**Comment by atherdon** [comment]
теперь мы будем делать расписание для этих групп.
в отдельном файле сгенерируй мавссив, внутри которого будет 3 месяца - сентябрь, октябрь, ноябрь. каждый месяц это тоже массив. предложи варианты, как сделать так, чтобы даты внутри массива, совпадали с днями недели?

**Comment by atherdon** [comment]
да, но у каждого месяца есть еще каледарные дни. как думаешь это решить


On 09-Jul-2018 at 07:27 PM, MinkoJ2 wrote:


Массив Месяц внутри которого 3 месяца

var september = ['September'];
var october = ['October'];
var november = ['November'];
var months = [september, october, november];

alert(months[0])

------------------------------
До статьи "Дата и время" ещё не дошёл, идей нет

@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403593130>,
or mute the thread

**Comment by MinkoJ2** [comment]
Что должно находится в массиве месяца?
@atherdon

**Comment by atherdon** [comment]
календарные дни и дни недели. ну представь себе календарик на месяц - что
обычно там внутри. там двухмерный массив - недели и дни :)


On 09-Jul-2018 at 08:17 PM, MinkoJ2 wrote:


Что должно находится в массиве месяца?
@atherdon <https://github.com/atherdon>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403606468>,
or mute the thread

**Comment by atherdon** [comment]
как пример календаря - можешь посмотреть тут как он выглядит: https://cool-features.netlify.com/Home

**Comment by MinkoJ2** [comment]
### Сделал запросы группы, и числа (либо дня недели) для вывода расписания
---
**[Click](https://github.com/MinkoJ2/Related-To-Testing-Article/tree/master/Objects) - schedule.js**

@atherdon

**Comment by atherdon** [comment]
videos, related to github - how they're going?

**Comment by MinkoJ2** [comment]
На смотрел ещё их

**Comment by atherdon** [comment]
learn english, view videos about git, read articles about project, read
about js in parallel, please. because all of this is important and we need
it


On 11-Jul-2018 at 12:49 PM, MinkoJ2 wrote:


На смотрел ещё их

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-404157091>,
or mute the thread

**Comment by MinkoJ2** [comment]
> @atherdon  
теперь мы будем делать расписание для этих групп.
в отдельном файле сгенерируй мавссив, внутри которого будет 3 месяца - сентябрь, октябрь, ноябрь. каждый месяц это тоже массив. предложи варианты, как сделать так, чтобы даты внутри массива, совпадали с днями недели?

**Did this task - [Click](https://github.com/MinkoJ2/Related-To-Testing-Article/tree/master/Objects)
Look schedule_Al.js**
@atherdon

**Comment by atherdon** [comment]
ну в shedule нормально ты извратился, мне нравится. чувствуется что "страдал" пока делал.
конечно теперь надо чтобы разузнал как работает moment.js - и еше поучил одну из двух библиотек - lodash/underscore.

оба этих дейтсвия упростили бы задачу и уменьшили количество строк. но вообще норм. теперь бы один большой объект по колледжу увидеть, как тут https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-403377920
только чтобы работало

**Comment by MinkoJ2** [comment]
Ты хочешь чтобы я закинул расписание в объект колледжа?????

**Comment by atherdon** [comment]
нет, я хочу большой объект колледж. без того что ты разбиваешь его на
подмассивы. и чтобы каждый ученик был отдельным объектом. короче сделай
хорошо :)


On 16-Jul-2018 at 06:56 AM, Edward wrote:


Ты хочешь чтобы я закинул расписание в объект колледжа?????

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/MinkoJ2/Related-To-Testing-Article/issues/2#issuecomment-405162159>,
or mute the thread

**Comment by MinkoJ2** [comment]
**[Смотри файлы groups.js, schedule.js, and college.js](https://github.com/MinkoJ2/Related-To-Testing-Article/tree/master/Objects)**
